include/LightGBM/utils/ChunkedArray.hpp

```diff
@@ -0,0 +1,260 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *
+ * Author: Alberto Ferreira
+ */
+#ifndef LIGHTGBM_UTILS_CHUNKED_ARRAY_H_
+#define LIGHTGBM_UTILS_CHUNKED_ARRAY_H_
+
+#include <LightGBM/utils/log.h>
+
+#include <stdint.h>
+
+#include <new>
+#include <vector>
+#include <algorithm>
```

Comment:
We try to keep includes in alphabetical order for better readability and not to include the same element twice in case of huge list.
```suggestion
#include <algorithm>
#include <new>
#include <vector>
```